torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -389,7 +393,8 @@ void pushProfilingCallbacks(const std::unordered_set<at::RecordScope>& scopes) {
             }
           }
 
-          state_ptr->reportClientActivity(fn, kineto_ctx_ptr);
+          kineto_ctx_ptr->endUS = getTimeUs();
```

Comment:
Why move this here?

Reply:
This is so that `reportClientActivity` can assume the start and end time are already encoded in ctx. When backend's report the runtime per op they will use indirectly use `reportClientActivity`. At the time of recording runtime of op, backends already have the duration/start and end time.